src/Components/Search/NewSearch/NewSearchBarInput.tsx

```diff
@@ -10,92 +9,213 @@ import { NewSearchBarInput_viewer$data } from "__generated__/NewSearchBarInput_v
 }) => {
+  const tracking = useTracking()
   const { t } = useTranslation()
-  const [value, setValue] = useState("")
+  const [value, setValue] = useState(getSearchTerm(window.location))
+  const [selectedPill, setSelectedPill] = useState<PillType>(TOP_PILL)
+  const [fetchCounter, setFetchCounter] = useState(0)
+  const { router } = useRouter()
 
   const options = extractNodes(viewer.searchConnection)
-  const formattedOptions = options.map(option => {
-    return {
-      text: option.displayLabel!,
-      value: option.displayLabel!,
-      subtitle: getLabel({
-        displayType: option.displayType ?? "",
+  const formattedOptions: SuggionItemOptionProps[] = options.map(
+    (option, index) => {
+      return {
+        text: option.displayLabel!,
+        value: option.displayLabel!,
+        subtitle:
+          getLabel({
+            displayType: option.displayType ?? "",
+            typename: option.__typename,
+          }) ?? "",
+        imageUrl: option.imageUrl!,
+        showArtworksButton: !!option.statuses?.artworks,
+        showAuctionResultsButton: !!option.statuses?.auctionLots,
+        href: option.href!,
         typename: option.__typename,
-      }),
-      imageUrl: option.imageUrl!,
-      showArtworksButton: !!option.statuses?.artworks,
-      showAuctionResultsButton: !!option.statuses?.auctionLots,
-      href: option.href!,
+        item_number: index,
+        item_type: option.displayType!,
+      }
     }
-  })
+  )
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setValue(event.target.value)
+  useUpdateEffect(() => {
+    tracking.trackEvent({
+      action_type:
+        options.length > 0
+          ? DeprecatedSchema.ActionType.SearchedAutosuggestWithResults
+          : DeprecatedSchema.ActionType.SearchedAutosuggestWithoutResults,
+      query: value,
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [fetchCounter])
+
+  const refetch = (value: string, entity?: string) => {
+    const entities = entity ? [entity] : []
+    const performanceStart = performance && performance.now()
 
-    //TODO: Add throttle
     relay.refetch(
       {
         hasTerm: true,
-        term: event.target.value,
+        term: value,
+        entities: entities,
       },
       null,
       error => {
         if (error) {
           logger.error(error)
           return
         }
+
+        if (performanceStart && getENV("VOLLEY_ENDPOINT")) {
+          reportPerformanceMeasurement(performanceStart)
+        }
+
+        setFetchCounter(prevCounter => prevCounter + 1)
       }
     )
   }
 
+  const debouncedSearchRequest = useMemo(
+    () => debounce(refetch, SEARCH_DEBOUNCE_DELAY),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  const clearSearchInput = () => {
+    setValue("")
+  }
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value)
+
+    if (event.target.value.length > 1)
```

Comment:
**suggestion**: create a helper method so that there is one source of truth

```javascript
const NewSearchBarInput: FC<NewSearchBarInputProps> = ({ ... }) => {
  ...
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const newValue = event.target.value
    
    setValue(newValue)

    if (shouldStartSearching(newValue))
      debouncedSearchRequest(event.target.value)
  }

  ...DESKTOP_NAV_BAR_TOP_TIER_HEIGHT

  return (
    <AutocompleteInput
      ...
      options={shouldStartSearching(value) ? formattedOptions : {}}
      ...
    />
  )
}

const shouldStartSearching = (value: string) => {
  return value.length >= 2
}
```